src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -968,7 +968,7 @@ private bool HasCovariantArrayConversion(TypeSymbol source, TypeSymbol destinati
             // * Both SE and TE are reference types.
             // * An implicit reference conversion exists from SE to TE.
             return
-                (s.Rank == d.Rank) &&
+                (s.Rank == d.Rank && s.IsSZArray == d.IsSZArray) &&
```

Comment:
It seems like we use this pattern a lot in the code base.  Would it make sense to capture this in a method?  Something like IsMatchingRankAndType?